compiler/src/typed/env.re

```diff
@@ -2036,8 +2036,9 @@ let add_components =
             Tbl.add(m, x, acc);
           } else {
             assert(
+              // TODO: Shouldn't this be a `try`?
```

Comment:
@ospencer @belph what's going on with this code?

Reply:
Looks like it came from this commit (which you made): https://github.com/grain-lang/grain/pull/317/commits/c3133d25a797bfae252e0f8d6a6107f5e35a90ab

What led to you needing to make the change?

Reply:
Yes, I added this comment because I needed you and @ospencer to look at the code right below it. When you upgraded the Reason to latest master, it removed the parens surrounding the annotated type and forced me to look at this code to figure out what was going on.

As you'll notice, this code is using a `switch` statement with a catch-all in the first match, so nothing else will fallthrough. Further, you'll see there is a pattern match that is trying to match an exception, which I'm pretty sure only works with a `try` statement, and not a `switch` statement.

So I wanted you both to look at it and figure out if there are other problems in this code, and I added the TODO as a point of reference in this PR.

Reply:
You can catch exceptions with `switch`/`match` statements like this. In this case since there's a wildcard branch, it's just acting like a regular `try`. It's a bit different though since with a try, you'd have to write it like this to be equivalent:

```re
try (IdTbl.find_name(...); true) { _ => false }
```

Reply:
or actually
```re
try (ignore(IdTbl.find_name(...)); true) { _ => false }
```

Reply:
Yeah, what he said :)
So, @phated, are we good here?

Reply:
I think we should just remove the TODOs

Reply:
Done.